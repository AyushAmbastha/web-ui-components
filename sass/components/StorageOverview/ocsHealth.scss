.ocs-health__body {
  max-height: 130px;
}

.ocs-health__icon--error {
  font-size: 1.333em;
  color: red;
  line-height: 1.333em;
}

.ocs-health__icon--ok {
  font-size: 1.333em;
  color: yellowgreen;
  line-height: 1.333em;
}

.ocs-health__icon--warning {
  font-size: 1.333em;
  color: goldenrod;
  line-height: 1.333em;
}

.ocs-ocshealth__row-status-item-text {
  margin-left: 1em;
  margin-top: 0.3em;
  font-size: 1.333em;
}

.ocs-ocshealth__row-status-item {
  display: flex;
  margin-right: 0.5em;
  flex-shrink: 0;
}
