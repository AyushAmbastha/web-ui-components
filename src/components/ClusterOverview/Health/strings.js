export const HEALTHY = 'Cluster is Healthy';
export const ERROR = 'Cluster health is in Error State';
export const UNKNOWN = 'Not available';
