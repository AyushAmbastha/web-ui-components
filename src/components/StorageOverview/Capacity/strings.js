export const TOTAL = 'Total Capacity';
export const CAPACITY = 'Requested vs Used';
export const VMSVSPODS = 'VMs vs Pods';
