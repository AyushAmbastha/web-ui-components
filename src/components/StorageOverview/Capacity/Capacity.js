import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'patternfly-react';

import { InlineLoading } from '../../Loading';

import {
  DashboardCard,
  DashboardCardBody,
  DashboardCardHeader,
  DashboardCardTitle,
} from '../../Dashboard/DashboardCard';
import { Dropdown } from '../../Form/Dropdown';
import { DashboardCardActionsBody } from '../../Dashboard/DashboardCard/DashboardCardActionsBody';

import { StorageOverviewContext } from '../StorageOverviewContext';
import { CapacityItem } from '../../Dashboard/Capacity/CapacityItem';
import { formatBytes } from '../../../utils';
import { getCapacityStats } from '../../../selectors';
import { CapacityBody } from '../../Dashboard/Capacity/CapacityBody';
import { TOTAL, CAPACITY, VMSVSPODS } from './strings';

const capacityTypes = [TOTAL, CAPACITY, VMSVSPODS];

export class Capacity extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      capacity: capacityTypes[0],
    };
  }

  convertMetrics = (total, used, title) => ({ total, used, title });

  changeCapacityMetric = newVal => {
    this.setState({ capacity: newVal }, this.getNewCapacityMetric);
  };

  getNewCapacityMetric = () => {
    switch (this.state.capacity) {
      case TOTAL:
        return this.convertMetrics(this.props.capacityTotal, this.props.capacityUsed, TOTAL);
      case CAPACITY:
        return this.convertMetrics(this.props.capacityRequested, this.props.capacityUsed, CAPACITY);
      case VMSVSPODS:
        return this.convertMetrics(this.props.podsCapacity, this.props.vmsCapacity, VMSVSPODS);
      default:
        return [];
    }
  };

  render() {
    const { capacityTotal, capacityRequested, capacityUsed, vmsCapacity, podsCapacity, LoadingComponent } = this.props;

    const capacitymetric = this.getNewCapacityMetric();
    const isLoading = !capacityTotal && !capacityRequested && !capacityUsed && !vmsCapacity && !podsCapacity;

    return (
      <DashboardCard className="kubevirt-capacity__card">
        <DashboardCardHeader>
          <Row>
            <Col lg={6} md={6} sm={6} xs={6}>
              <DashboardCardTitle>Capacity</DashboardCardTitle>
            </Col>
            <Col>
              <DashboardCardActionsBody>
                <Dropdown
                  id="capacity-type"
                  value={this.state.capacity}
                  onChange={this.changeCapacityMetric}
                  choices={capacityTypes}
                  disabled={isLoading}
                  groupClassName="kubevirt-dropdown__group"
                />
              </DashboardCardActionsBody>
            </Col>
          </Row>
        </DashboardCardHeader>
        <DashboardCardBody>
          <CapacityBody>
            <CapacityItem
              id="capacity"
              title={capacitymetric.title}
              used={getCapacityStats(capacitymetric.used)}
              total={getCapacityStats(capacitymetric.total)}
              formatValue={formatBytes}
              isLoading={!(capacitymetric.used && capacitymetric.total)}
              LoadingComponent={LoadingComponent}
            />
          </CapacityBody>
        </DashboardCardBody>
      </DashboardCard>
    );
  }
}

Capacity.defaultProps = {
  capacityTotal: null,
  capacityRequested: null,
  capacityUsed: null,
  vmsCapacity: null,
  podsCapacity: null,
  LoadingComponent: InlineLoading,
};

Capacity.propTypes = {
  capacityTotal: PropTypes.object,
  capacityRequested: PropTypes.object,
  capacityUsed: PropTypes.object,
  vmsCapacity: PropTypes.object,
  podsCapacity: PropTypes.object,
  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
};

export const CapacityConnected = () => (
  <StorageOverviewContext.Consumer>
    {props => (
      <Capacity
        capacityTotal={props.capacityTotal}
        capacityUsed={props.capacityUsed}
        capacityRequested={props.capacityRequested}
        vmsCapacity={props.vmsCapacity}
        podsCapacity={props.podsCapacity}
        LoadingComponent={props.LoadingComponent}
      />
    )}
  </StorageOverviewContext.Consumer>
);

CapacityConnected.propTypes = {
  ...Capacity.propTypes,
};

CapacityConnected.defaultProps = {
  ...Capacity.defaultProps,
};
