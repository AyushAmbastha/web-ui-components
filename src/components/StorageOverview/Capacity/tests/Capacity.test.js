import React from 'react';
import { render, shallow, mount } from 'enzyme';

import { Capacity, CapacityConnected } from '../Capacity';
import { default as CapacityFixtures } from '../fixtures/Capacity.fixture';
import { default as StorageOverviewFixtures } from '../../fixtures/StorageOverview.fixture';
import { StorageOverviewContext } from '../../StorageOverviewContext';
import { selectDropdownItem } from '../../../../tests/enzyme';

import { TOTAL, CAPACITY } from '../strings';

// eslint-disable-next-line react/prop-types
const testCapacityOverview = ({ props }) => <Capacity {...props} />;

const getCapacityDropdown = component => component.find('#capacity-type');

const testCapacityResults = (component, capacity) => {
  selectDropdownItem(getCapacityDropdown(component), capacity);
  expect(component.state().capacity).toBe(capacity);
};

describe('<Capacity />', () => {
  CapacityFixtures.forEach(fixture => {
    it(`renders ${fixture.name} correctly`, () => {
      const component = shallow(testCapacityOverview(fixture));
      expect(component).toMatchSnapshot();
    });
  });
  it('switches between capacity dropdown options', () => {
    const component = mount(testCapacityOverview(CapacityFixtures[0]));
    expect(component.state().capacity).toBe(TOTAL);
    testCapacityResults(component, CAPACITY);
  });
  it('renders correctly with Provider', () => {
    const component = render(
      <StorageOverviewContext.Provider value={StorageOverviewFixtures[0].props}>
        <CapacityConnected />
      </StorageOverviewContext.Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
