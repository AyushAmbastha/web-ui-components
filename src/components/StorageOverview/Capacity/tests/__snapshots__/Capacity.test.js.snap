// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Capacity /> renders Capacity correctly 1`] = `
<DashboardCard
  className="kubevirt-capacity__card"
>
  <DashboardCardHeader
    className={null}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        lg={6}
        md={6}
        sm={6}
        xs={6}
      >
        <DashboardCardTitle
          className={null}
        >
          Capacity
        </DashboardCardTitle>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
      >
        <DashboardCardActionsBody>
          <Dropdown
            choices={
              Array [
                "Total Capacity",
                "Requested vs Used",
                "VMs vs Pods",
              ]
            }
            disabled={false}
            groupClassName="kubevirt-dropdown__group"
            id="capacity-type"
            onChange={[Function]}
            value="Total Capacity"
            withTooltips={false}
          />
        </DashboardCardActionsBody>
      </Col>
    </Row>
  </DashboardCardHeader>
  <DashboardCardBody
    LoadingComponent={[Function]}
    isLoading={false}
  >
    <CapacityBody>
      <CapacityItem
        LoadingComponent={[Function]}
        formatValue={[Function]}
        id="capacity"
        isLoading={false}
        title="Total Capacity"
        total={11}
        used={5}
      />
    </CapacityBody>
  </DashboardCardBody>
</DashboardCard>
`;

exports[`<Capacity /> renders Loading Capacity correctly 1`] = `
<DashboardCard
  className="kubevirt-capacity__card"
>
  <DashboardCardHeader
    className={null}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        lg={6}
        md={6}
        sm={6}
        xs={6}
      >
        <DashboardCardTitle
          className={null}
        >
          Capacity
        </DashboardCardTitle>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
      >
        <DashboardCardActionsBody>
          <Dropdown
            choices={
              Array [
                "Total Capacity",
                "Requested vs Used",
                "VMs vs Pods",
              ]
            }
            disabled={true}
            groupClassName="kubevirt-dropdown__group"
            id="capacity-type"
            onChange={[Function]}
            value="Total Capacity"
            withTooltips={false}
          />
        </DashboardCardActionsBody>
      </Col>
    </Row>
  </DashboardCardHeader>
  <DashboardCardBody
    LoadingComponent={[Function]}
    isLoading={false}
  >
    <CapacityBody>
      <CapacityItem
        LoadingComponent={[Function]}
        formatValue={[Function]}
        id="capacity"
        isLoading={true}
        title="Total Capacity"
        total={null}
        used={null}
      />
    </CapacityBody>
  </DashboardCardBody>
</DashboardCard>
`;

exports[`<Capacity /> renders correctly with Provider 1`] = `
<div
  class="card-pf kubevirt-dashboard__card kubevirt-capacity__card"
>
  <div
    class="card-pf-heading kubevirt-dashboard__card-heading"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-6 col-md-6 col-sm-6 col-xs-6"
      >
        <h2
          class="card-pf-title kubevirt-dashboard__card-title"
        >
          Capacity
        </h2>
      </div>
      <div
        class=""
      >
        <div
          class="kubevirt-dashboard__actions-body"
        >
          <div
            class="kubevirt-dropdown__group btn-group btn-group-justified"
          >
            <div
              class="dropdown btn-group btn-group-default"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="kubevirt-dropdown dropdown-toggle btn btn-default"
                id="capacity-type"
                role="button"
                type="button"
              >
                Total Capacity 
                <span
                  class="caret"
                />
              </button>
              <ul
                aria-labelledby="capacity-type"
                class="dropdown-menu"
                role="menu"
              >
                <li
                  class=""
                  role="presentation"
                >
                  <a
                    href="#"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Total Capacity
                  </a>
                </li>
                <li
                  class=""
                  role="presentation"
                >
                  <a
                    href="#"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Requested vs Used
                  </a>
                </li>
                <li
                  class=""
                  role="presentation"
                >
                  <a
                    href="#"
                    role="menuitem"
                    tabindex="-1"
                  >
                    VMs vs Pods
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card-pf-body"
  >
    <div
      class="kubevirt-capacity__items"
    >
      <div
        class="kubevirt-capacity__item"
      >
        <div
          class="kubevirt-capacity__item-title"
        >
          Total Capacity
        </div>
        <h6
          class="kubevirt-capacity__item-description"
        >
          6 B available out of 11 B
        </h6>
        <div
          class="kubevirt-capacity__graph"
          id="donut-chart-capacity"
        >
          <svg
            viewBox="0 0 200 200"
          >
            <g>
              <path
                d="M1.0378342131579386,-87.99387990164998A88,88,0,0,1,25.653005755549977,84.17792641604876L23.41115576348232,76.49782863466105A80,80,0,0,0,1.0378342131579519,-79.99326784265035Z"
                role="presentation"
                shape-rendering="auto"
                style="fill:#06c;padding:8px;stroke:transparent;stroke-width:1"
                transform="translate(100, 100)"
              />
              <path
                d="M23.66049113813229,84.75954907325996A88,88,0,1,1,-1.0378342131579104,-87.99387990164998L-1.0378342131578906,-79.99326784265035A80,80,0,1,0,21.41864114606461,77.07945129187226Z"
                role="presentation"
                shape-rendering="auto"
                style="fill:#bee1f4;padding:8px;stroke:transparent;stroke-width:1"
                transform="translate(100, 100)"
              />
            </g>
            <text
              direction="inherit"
              dx="0"
              dy="8.875"
              x="100"
              y="88"
            >
              <tspan
                dx="0"
                style="font-size:25px;fill:#252525;font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;stroke:transparent"
                text-anchor="middle"
                x="100"
              >
                45%
              </tspan>
            </text>
            <text
              direction="inherit"
              dx="0"
              dy="5.324999999999999"
              x="100"
              y="112"
            >
              <tspan
                dx="0"
                style="font-size:15px;fill:#252525;font-family:'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif;stroke:transparent"
                text-anchor="middle"
                x="100"
              >
                Used
              </tspan>
            </text>
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
